cargo-spatial/src/codegen.rs

```diff
@@ -0,0 +1,127 @@
+use crate::config::Config;
+use log::*;
+use spatialos_sdk_code_generator::{generator, schema_bundle};
+use std::ffi::OsString;
+use std::fs::{self, File};
+use std::io::prelude::*;
+use std::path::*;
+use std::process::Command;
+use tap::*;
+
+/// Performs code generation for the project described by `config`.
+///
+/// Assumes that the current working directory is the root directory of the project,
```

Comment:
Should we check or assert against this at the start of the function?

Reply:
Yeah, I think that's a good idea. There's a bit more to this that we might want to support, i.e. traversing up the directory hierarchy to find a `Spatial.toml` file in order to support running `cargo spatial` commands from sub-directories of the project (which is something Cargo supports), but I think for now it's a good idea to add asserts around the invariants we currently have.